### ISSUE TITLE
Check if docker is installed before starting the Apps

### ISSUE THREAD

**Issue by apoorvsadana** [keep-starknet-strange/madara-tsukuyomi#19]
Right now, the Apps section assume Docker is already installed and try to pull and run the app if the user interacts with it. However, this maybe not be true since the people using the app are non devs. Hence, we want a feature to check if Docker is installed or not and accordingly install it.

User flow

1. User clicks on the `Apps` section
2. `renderer` calls `main` to check if docker is installed
    - YES: move to step 3
    - NO: Give a message in full screen saying `Docker` is not installed and a once click button which installs it and shows the progress while installing. Start the docker daemon after installation.
3. `renderer` calls `main` to check if daemon is running
    - YES: move to step 4
    - NO: Start the daemon
4. Show the apps screen

Currently, we go directly from 1 -> 4 so the apps break if docker is not installed or the daemon isn't running.

Notes
 - In linux, starting the daemon is a simple one line command
 - In mac, starting the daemon isn't very simple and we can ideally use `colima` to abstract the complications. You can read more about that [here](https://apple.stackexchange.com/questions/373888/how-do-i-start-the-docker-daemon-on-macos)
 - We will need to check how it works on Windows
 
 Design
- The design shouldn't be complicated for this and we can take inference from already existing components. If needed, I can make a sample screen as well.

**Comment by NikolaGrujicic** [comment]
i will take this

**Comment by TedZy829** [comment]
may I know if this task is done already or is it still open for contributors?

**Comment by apoorvsadana** [comment]
It's open! @NikolaGrujicic is working on another task atm so you can pick this if you like

**Comment by apoorvsadana** [comment]
Let me know if you're picking this up @TedZy829, I will assign it to you

**Comment by TedZy829** [comment]
oh okay, is there any estimation on how long to finish this task?

**Comment by apoorvsadana** [comment]
@TedZy829 that would depend on your familiarity with electron and the app itself. Since this is your first issue it will take longer than usual as well. If you want to pick something smaller, I would suggest trying some of the other good first issues :). They are smaller than this one.

**Comment by Rajdeep1311** [comment]
I guess writing a shell script might do the job, if docker is installed it won't install otherwise it would.

**Comment by 0xGavinOwen** [comment]
can i take this issue?

**Comment by apoorvsadana** [comment]
Hey @Rajdeep1311, sorry for the late reply. We want the logic of checking installed/not installed to be packaged within the code itself.

**Comment by apoorvsadana** [comment]
@Rajdeep1311 are you interested in picking up an issue?
